dapr/cli
# Also get version from helm chart

# Description

1. We can now fallback and read the helm index.yaml (which is statically served from a CDN) if we are unable to reach the github API for whatever reason.
2. Include an error that tells users how to work around the github API being down.
3. We print a warning if we can't install the dashboard, but allow install to continue.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #677
Also close: #755
Also close: #747

## Manual Test

```bash
 cli % for i in `seq 1 60`; do curl https://api.github.com/repos/dapr/dapr/releases  >/dev/null 2>&1 &  done
 cli % ./dist/darwin_amd64/release/dapr init
⌛  Making the jump to hyperspace...
⚠  Failed to get runtime version: 'https://api.github.com/repos/dapr/dapr/releases - 403 Forbidden'. Trying secondary source
ℹ️  Installing runtime version 1.2.2
⚠  cannot get the latest dashboard version: 'https://api.github.com/repos/dapr/dashboard/releases - 403 Forbidden'. Try specifying --dashboard-version=<desired_version>
⚠  continuing, but dashboard will be unavailable
↘  Downloading binaries and setting up components... 
Dapr runtime installed to /Users/willardstanley/.dapr/bin, you may run the following to add it to your path if you want to run daprd directly:
    export PATH=$PATH:/Users/willardstanley/.dapr/bin
✅  Downloading binaries and setting up components...
✅  Downloaded binaries and completed components set up.
ℹ️  daprd binary has been installed to /Users/willardstanley/.dapr/bin.
ℹ️  dapr_placement container is running.
ℹ️  dapr_redis container is running.
ℹ️  dapr_zipkin container is running.
ℹ️  Use `docker ps` to check running containers.
✅  Success! Dapr is up and running. To get started, go here: https://aka.ms/dapr-getting-started
```

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Created/updated tests
* [ ] Extended the documentation
